internal/translator/translate.go

```diff
@@ -143,12 +144,21 @@ func translateRoute(irRoute irv1beta1.Route, routeLCP string) (hpv1.Route, []str
 
 		}
 		if irService.HealthCheck != nil {
-			route.HealthCheckPolicy = &hpv1.HTTPHealthCheckPolicy{
-				Path:                    irService.HealthCheck.Path,
-				Host:                    irService.HealthCheck.Host,
-				TimeoutSeconds:          irService.HealthCheck.TimeoutSeconds,
-				UnhealthyThresholdCount: irService.HealthCheck.UnhealthyThresholdCount,
-				HealthyThresholdCount:   irService.HealthCheck.HealthyThresholdCount,
+			if seenHealthCheckPolicy == "" {
+				// Copy the first strategy we encounter into the HP loadbalancerpolicy
+				// and save that we've seen that one.
+				seenHealthCheckPolicy = irService.HealthCheck.Host + irService.HealthCheck.Path
+				route.HealthCheckPolicy = &hpv1.HTTPHealthCheckPolicy{
+					Path:                    irService.HealthCheck.Path,
+					Host:                    irService.HealthCheck.Host,
+					TimeoutSeconds:          irService.HealthCheck.TimeoutSeconds,
+					UnhealthyThresholdCount: irService.HealthCheck.UnhealthyThresholdCount,
+					HealthyThresholdCount:   irService.HealthCheck.HealthyThresholdCount,
+				}
+			} else {
+				if seenHealthCheckPolicy != irService.HealthCheck.Host+irService.HealthCheck.Path {
```

Comment:
Maybe it would be more predictable to either compare all the fields (so we can guarantee that they are actually the same) or none of them (so we issue the warning based on the textual presence of the additional checks). This seems like a slightly uncomfortable middle ground between these two poles.

Reply:
yeah, you're right. I guess I'll do the compare for the whole thing and use just those two fields for logging only.